bootloader/lkenv/lkenv.go

```diff
@@ -81,142 +159,117 @@ func copyString(b []byte, s string) {
 	}
 }
 
-func NewEnv(path string) *Env {
-	return &Env{
+func NewEnv(path string, version Version) *Env {
+	e := &Env{
 		path:    path,
 		pathbak: path + "bak",
-		env: SnapBootSelect_v1{
+		version: version,
+	}
+
+	switch version {
+	case V1:
+		// TODO: move the initial assignment of members to setup() ?
+		e.variant = &SnapBootSelect_v1{
 			Signature: SNAP_BOOTSELECT_SIGNATURE,
-			Version:   SNAP_BOOTSELECT_VERSION,
-		},
+			Version:   SNAP_BOOTSELECT_VERSION_V1,
+		}
+	case V2Recovery:
+		e.variant = &SnapBootSelect_v2_recovery{
+			Signature: SNAP_BOOTSELECT_SIGNATURE,
```

Comment:
should the signature for this be same? they don't quite carry the same kind of values  (kernels vs recovery systems)

Reply:
as discussed, this will be different, after speaking with @kubiko we decided to use with `SRse`, which is `S(Snap)R(Recovery)select(se)` like the boot environment signature select comes from `SBse` which was `Snap(S)B(Boot)select(se)` :smile: 

For posterity's sake, the bootloader doesn't currently have any attachment to a particular value for this, the bootloader patches can be adjusted as needed.